shared/test/toc_test.js

```diff
@@ -0,0 +1,65 @@
+/* Any copyright is dedicated to the Public Domain.
+ * http://creativecommons.org/publicdomain/zero/1.0/ */
+
+describe("loop.TableOfContents", () => {
+  "use strict";
+
+  let { expect } = chai;
+  let TestUtils = React.addons.TestUtils;
+
+  let sandbox;
+  let dispatcher, participantStore;
+
+  beforeEach(() => {
+    sandbox = LoopMochaUtils.createSandbox();
+
+    dispatcher = new loop.Dispatcher();
+    sandbox.stub(dispatcher, "dispatch");
+
+    participantStore = new loop.store.ParticipantStore(dispatcher);
+  });
+
+  afterEach(() => {
+    sandbox.restore();
+    LoopMochaUtils.restore();
+  });
+
+  describe("ToC.RoomPresenceView", () => {
+    let view, participant;
+
+    function mountTestComponent() {
+      let participants = new Map();
+      participants.set("fakeID", participant);
+
+      participantStore.setStoreState({
+        participants
+      });
```

Comment:
We should probably use the ParticipantStore APIs of `updatedParticipant` and `updatedPresence` to avoid reaching into the internals of a separate class.